addons/dev-helpers/setup-go-env.sh

```diff
@@ -50,19 +50,6 @@ else
     install
 fi
 
-log_section "Setup variables and directories for Golang environment"
-# we are in a packer build
-if [ -n "$PACKER_BUILD_NAME" ]; then
-    declare -p GOPATH GO_REPO
-    mkdir -v -p $GOPATH/src/$GO_REPO
-else
-    setup
-    declare -p GOPATH GO_REPO
-    mkdir -v -p $GOPATH/src/$GO_REPO
-    if [ -d $GOPATH/src/$GO_REPO/go ]; then
-        die "Directory $GOPATH/src/$GO_REPO/go already exists, cannot symlink it to /usr/local/pf/go"
-    else
-        ln -s -v /usr/local/pf/go $GOPATH/src/$GO_REPO/go
-    fi
-fi
+cd /usr/local/pf/go
```

Comment:
- You expect to be on a system where PF is installed. When building images, PF is not installed and I ẁill only copy `go.mod` file in temp location. We should use an environment variable here with a default value that can be override. We could still use `GO_REPO`
- Some parts of the script can be drop because they will not be used.